openml/runs/run.py

```diff
@@ -222,50 +249,84 @@ def get_metric_fn(self, sklearn_fn, kwargs={}):
             predictions_file_url = openml._api_calls._file_id_to_url(
                 self.output_files['predictions'], 'predictions.arff',
             )
-            predictions_arff = arff.loads(openml._api_calls._read_url(predictions_file_url))
+            predictions_arff = \
+                arff.loads(openml._api_calls._read_url(predictions_file_url))
             # TODO: make this a stream reader
         else:
-            raise ValueError('Run should have been locally executed or contain outputfile reference.')
+            raise ValueError('Run should have been locally executed or '
+                             'contain outputfile reference.')
+
+        # Need to know more about the task to compute scores correctly
+        task = get_task(self.task_id)
 
         attribute_names = [att[0] for att in predictions_arff['attributes']]
-        if 'correct' not in attribute_names:
-            raise ValueError('Attribute "correct" should be set')
-        if 'prediction' not in attribute_names:
-            raise ValueError('Attribute "predict" should be set')
+        if task.task_type == 'Supervised Classification' and 'correct' not in \
```

Comment:
here again references to the task name and the `correct` label. The first can be fixed easily. The second relies on a similar naming as in `generate_arff_dict`. As mentioned before, we should think of unifying these.

Reply:
Agreed. But again, this change seems bigger than the regression PR. Maybe do one thing at a time?